### PR TITLE
[CLOUDTRUST-2164] Fix access control on child objects

### PR DESCRIPTION
**Pull request by harture**

### CODE REVIEW

File: keycloak-rest-api-extensions/src/main/java/io/cloudtrust/keycloak/services/resource/api/account/FixedAccountCredentialResource.java [hunk 1]
```diff
@@ -86,9 +88,15 @@ public FixedAccountCredentialResource(KeycloakSession session, EventBuilder even
...
     @NoCache
-    public void removeCredential(final @PathParam("credentialId") String credentialId) {
+    public Response removeCredential(final @PathParam("credentialId") String credentialId) {
         auth.require(AccountRoles.MANAGE_ACCOUNT);
+
+        if (session.userCredentialManager().getStoredCredentialById(realm, user, credentialId) == null) {
+            return Response.status(NOT_FOUND).build();
+        }
+
         session.userCredentialManager().removeStoredCredential(realm, user, credentialId);
```
Comment (fperot74) [inline keycloak-rest-api-extensions/src/main/java/io/cloudtrust/keycloak/services/resource/api/account/FixedAccountCredentialResource.java]: Can't you simply check if removeStoredCredential returns true or false?

File: keycloak-rest-api-extensions/src/main/java/io/cloudtrust/keycloak/services/resource/api/account/FixedAccountCredentialResource.java [hunk 2]
```diff
@@ -97,9 +105,15 @@ public void removeCredential(final @PathParam("credentialId") String credentialI
...
     @Path("{credentialId}/label")
-    public void setLabel(final @PathParam("credentialId") String credentialId, String userLabel) {
+    public Response setLabel(final @PathParam("credentialId") String credentialId, String userLabel) {
         auth.require(AccountRoles.MANAGE_ACCOUNT);
+
+        if (session.userCredentialManager().getStoredCredentialById(realm, user, credentialId) == null) {
+            return Response.status(NOT_FOUND).build();
+        }
+
         session.userCredentialManager().updateCredentialLabel(realm, user, credentialId, userLabel);
```
Comment (fperot74) [inline keycloak-rest-api-extensions/src/main/java/io/cloudtrust/keycloak/services/resource/api/account/FixedAccountCredentialResource.java]: Can't you simply check if updateCredentialLabel returns true or false?

File: keycloak-rest-api-extensions/src/main/java/io/cloudtrust/keycloak/services/resource/api/account/FixedAccountCredentialResource.java [hunk 3]
```diff
@@ -119,9 +133,19 @@ public void moveToFirst(final @PathParam("credentialId") String credentialId){
...
+
+        if (session.userCredentialManager().getStoredCredentialById(realm, user, credentialId) == null) {
+            return Response.status(NOT_FOUND).build();
+        }
+
+        if (newPreviousCredentialId != null && session.userCredentialManager().getStoredCredentialById(realm, user, newPreviousCredentialId) == null) {
+            return Response.status(NOT_FOUND).build();
+        }
+
         session.userCredentialManager().moveCredentialTo(realm, user, credentialId, newPreviousCredentialId);
```
Comment (fperot74) [inline keycloak-rest-api-extensions/src/main/java/io/cloudtrust/keycloak/services/resource/api/account/FixedAccountCredentialResource.java]: Same here

File: keycloak-rest-api-extensions/src/main/java/io/cloudtrust/keycloak/services/resource/api/account/FixedAccountCredentialResource.java [hunk 4]
```diff
@@ -231,4 +255,5 @@ public void setConfirmation(String confirmation) {
 
     }
 
+
```
Comment (fperot74) [inline keycloak-rest-api-extensions/src/main/java/io/cloudtrust/keycloak/services/resource/api/account/FixedAccountCredentialResource.java]: Is it a collection of blank lines? ;-)

File: keycloak-rest-api-extensions/src/main/java/io/cloudtrust/keycloak/services/resource/api/account/FixedAccountCredentialResource.java [hunk 5]
```diff
@@ -231,4 +255,11 @@ public void setConfirmation(String confirmation) {
 
     }
 
+    private boolean credentialOwnedByUser(RealmModel realm, UserModel user, String credentialId) {
+        List<CredentialModel> models = session.userCredentialManager().getStoredCredentials(realm, user);
+        if (models.stream().filter( c -> c.getId() == credentialId).count() == 0) {
```
Comment (fperot74) [inline keycloak-rest-api-extensions/src/main/java/io/cloudtrust/keycloak/services/resource/api/account/FixedAccountCredentialResource.java]: You can just `return models.stream().....count()==0;`

Comment (harture) [reply keycloak-rest-api-extensions/src/main/java/io/cloudtrust/keycloak/services/resource/api/account/FixedAccountCredentialResource.java]: Good remark ... :)

File: keycloak-rest-api-extensions/src/main/java/io/cloudtrust/keycloak/services/resource/api/account/FixedAccountCredentialResource.java [hunk 6]
```diff
@@ -231,4 +255,8 @@ public void setConfirmation(String confirmation) {
 
     }
 
+    private boolean credentialOwnedByUser(RealmModel realm, UserModel user, String credentialId) {
+        List<CredentialModel> models = session.userCredentialManager().getStoredCredentials(realm, user);
+        return models.stream().filter( c -> c.getId() == credentialId).count() == 0;
```
Comment (lagess) [inline keycloak-rest-api-extensions/src/main/java/io/cloudtrust/keycloak/services/resource/api/account/FixedAccountCredentialResource.java]: Should be == 1

**Pull request merged**